python/graphstorm/eval/evaluator.py

```diff
@@ -802,14 +804,16 @@ class GSgnnMrrLPEvaluator(GSgnnLPEvaluator):
     early_stop_strategy: str
         The early stop strategy. GraphStorm supports two strategies:
         1) consecutive_increase and 2) average_increase.
+    k: int
+        k used for computing metrics such as hits@k.
     """
-    def __init__(self, eval_frequency, data,
+    def __init__(self, eval_frequency, eval_metric, data,
                  num_negative_edges_eval, lp_decoder_type,
                  use_early_stop=False,
                  early_stop_burnin_rounds=0,
                  early_stop_rounds=3,
-                 early_stop_strategy=EARLY_STOP_AVERAGE_INCREASE_STRATEGY):
-        eval_metric = ["mrr"]
+                 early_stop_strategy=EARLY_STOP_AVERAGE_INCREASE_STRATEGY,
+                 k=100):
```

Comment:
It is better to define a hit@k evaluator.

Reply:
We have two choices:
1) define a GSgnnRankingLPEvaluator class and both GSgnnMrrLPEvaluator and GSgnnHitKLPEvaluator will inherate from GSgnnRankingLPEvaluator.
2) define a GSgnnRankingLPEvaluator class which implements mrr() and hit(), and rename it to GSgnnMrrLPEvaluator and  GSgnnRankingLPEvaluator.

I would prefer the first one.